### DISCUSSION TITLE
Режим сна на Pinecil от датчика холла

### DISCUSSION THREAD

**Discussion by Alexander290583** [Ralim/IronOS#1187] [Q&A]
Подскажите пожалуйста как доработать Pinecil чтобы он переключался в режим сна от датчика холла? Подскажите какой датчик использовать и куда в паять на плате?

**Comment by discip** [comment]
Good afternoon @Alexander290583,
it seems Russia is getting their hands on these devices 😁, since you are at least the 3rd person to start a conversation in Russian.
As @paulfertser and I asked these people to please use English to communicate here, I kindly ask you as well to do so.

About your issue:
Since I didn't need this feature yet, I was not even aware of the hall sensor not being installed on the board.
But here are some links and maybe @mikerocklewitz could help you, since he managed to get it to work as he stated [here](https://github.com/Ralim/IronOS/issues/1001#issuecomment-865995169).

https://files.pine64.org/doc/datasheet/pinecil/si7210-datasheet.pdf

kind regards

**Reply by mikerocklewitz** [reply]
The only issue i had was thinking it wasnt working when it was. You need a pretty strong neodymium magnet to trigger it through the housing. Honestly i dont even use it because having the magnet that close to the tip was annoying when placing it in the stand i use. I would always get stuck to the magnet.

As long as you match the part number to the schematic it should work out out of the gate.

My disclaimer is i have not tried any new firmware since whatever date that post by me was.

**Reply by River-Mochi** [reply]
@mikerocklewitz   
you do not necessarily need a strong magnet to get hall sensor to work. I got it to work with cheap neomydium 8mm magnets. I did a lot of testing with 4-5 different styles of stands and some different kinds of neomydium magnets.

you have to set  Hall to level 9 sensitivity in 2.17 firmware or  High sensitivity in 2.16 firmware (sleep menu).  I did many test and could share photos of how I put 1-2 small cheap 8mm round neomydium magnets on my stands and it worked.  The location on the stand is very convenient, you just put magnet on top right near where the Pinecil enters the stand.  but  I also put Pinecil into Debug mode (long press of [-]  button, then scroll with [+]  until you see Hall # .
Then put your pinecil in the stand and move the magnet around to a good spot where you have a high hall number and the Pinecil is sitting in the stand at the angle you like. 

Here in photo I have just 2 tiny cheap neomydium magnets (10 for $2.99)  and have full coverage of the pinecil to sleep when it enters the stand. 
If you do have a very expensive, and strong magnet you could put it underneath the stand but since I have tiny magnets I had to sit them on top to make the Pinecil work consistently.

**Reply by mikerocklewitz** [reply]
Ill do some more experimenting. I recently updated all my irons to the most current ironos.....i was setting one up for a gift and ended upgrading all of em.  I think i used it twice in the past 2 weeks since doing that.  

Got a t12/jbc clone controller im working on. Wireless paneldue mod im workin on. Monoprice mini klipper mod im workin on. On top of my main printer which is a never ending project. Recently added mini cluster to the list.

Lifes crazy. My kids n wife never leave me alone and my job that never wants me to leave 🤣  lately i just wanna solder 🤣🤣

**Reply by River-Mochi** [reply]
![2022-02-03 17 44 11](https://user-images.githubusercontent.com/97197236/153723817-6c1b9dbd-8f18-4024-81d3-fc8c139034ae.jpg)

I have 2 tiny magnets on top of the stand . the key for me is watching the Debug menu, Hall #  , and moving them to best spot for max coverage. 
If you updated to 2.17 then, there was a change to Hall sensor settings.   remember to go into Sleep, change Hall sensor setting,  9 is the most sensitive if you have smaller magnets and only can get around 25-50 reading.  I just do 9 unless the Pinecil sleeping too much and too much magnetized things on work area triggering it, then I change to level 8,  which is 50+ hall number.

**Comment by Alexander290583** [comment]
Good day! Thank you for your prompt response and for helping me get my bearings. All functions of this Pinecil. Sorry for my bad English, I don't know it, it's all Google translation. Thanks again for your active help, I don’t know if this function will be useful for me or maybe not. But whoever still useful information, it will come in handy. :)

**Reply by mikerocklewitz** [reply]
Its good to experiment.

Glad i could help!

**Reply by River-Mochi** [reply]
join the Pinecil discord and you will get more detailed help on  installing the hall sensor. 
**https://discord.gg/Qp78bDdP**

purchase this one, it is in schematic for pinecil.   SI7210-B-00-IV  from Silicon labs 
Look at U14  on this :  https://files.pine64.org/doc/Pinecil/Pinecil_schematic_v1.0a_20201120.pdf

**U14 empty before install**
![2022-02-01 03 16 04](https://user-images.githubusercontent.com/97197236/153313577-bc6ae6de-7fa8-4ab1-84a0-89361984a79f.jpg)


Installed Hall sensor  **SI7210-B-00-IV**
![2022-02-02 01 46 28](https://user-images.githubusercontent.com/97197236/153313646-9f4c34ec-c8e9-4ab7-82ea-f555385c736b.jpg)

**Join the Pine64 Discord chat**,  go to Pinecil channel, community volunteers more active on discord.  no one looks at the Github very much. there are also bridges to discord chat from Telegram/matrix.  see Pinecil Wiki for links.
